modules/core/src/main/java/org/apache/ignite/dump/DumpReader.java

```diff
@@ -135,13 +151,51 @@ public DumpReader(DumpReaderConfiguration cfg, IgniteLogger log) {
                                     return;
                                 }
 
-                                try (DumpedPartitionIterator iter = dump.iterator(node, grp, part)) {
+                                try (DumpedPartitionIterator iter = new DumpedPartitionIterator() {
+                                    /** */
+                                    final DumpedPartitionIterator delegate = dump.iterator(node, grp, part);
+
+                                    /** */
+                                    final AtomicBoolean consumerProcessingEntry = new AtomicBoolean(false);
+
+                                    /** {@inheritDoc } */
+                                    @Override public boolean hasNext() {
+                                        if (consumerProcessingEntry.compareAndSet(true, false))
+                                            statsLogger.recordProcessed();
+
+                                        return delegate.hasNext();
+                                    }
+
+                                    /** {@inheritDoc } */
+                                    @Override public DumpEntry next() {
+                                        if (consumerProcessingEntry.compareAndSet(true, false)) {
+                                            // Consumer didn't execute hasNext()
+                                            statsLogger.recordProcessed();
+                                        }
+
+                                        DumpEntry next = delegate.next();
+
+                                        consumerProcessingEntry.set(true);
+
+                                        return next;
+                                    }
+
+                                    /** {@inheritDoc } */
+                                    @Override public void close() throws Exception {
+                                        delegate.close();
+                                    }
+                                }) {
                                     if (log.isDebugEnabled()) {
                                         log.debug("Consuming partition [node=" + node + ", grp=" + grp +
                                             ", part=" + part + ']');
                                     }
 
                                     cnsmr.onPartition(grp, part, iter);
+
+                                    int partNo = partsProcessed.incrementAndGet();
+
+                                    if (log.isInfoEnabled())
+                                        log.info("Consumed partitions " + partNo + " of " + partsCnt);
```

Comment:
I see the pretty common log message in 188L. I suppose we need to log the partition counter at the same rate as entry counter. If user needs to log every partition he/she could use debug level of logging. WDYT?